src/test/java/org/logstash/beats/BeatsParserTest.java

```diff
@@ -50,14 +53,25 @@ public void testEncodingDecodingFields() {
         assertMessages(this.batch, decodedBatch);
     }
 
-
     @Test
-    public void  testCompressedEncodingDecodingFields() {
+    public void testCompressedEncodingDecodingFields() {
         this.batch.setProtocol(Protocol.VERSION_1);
         Batch decodedBatch = decodeCompressedBatch();
         assertMessages(this.batch, decodedBatch);
     }
 
+    @Test
+    public void testShouldNotCrashOnGarbageData() {
```

Comment:
I think it is fine to keep it together since it is so tightly coupled and it is just one exception.